kyo-core/shared/src/main/scala/kyo/StreamCoreExtensions.scala

```diff
@@ -189,6 +195,333 @@ object StreamCoreExtensions:
             f: Frame
         ): Stream[V, Abort[E] & S & Async] =
             other.mergeHaltingLeft(stream)(using i1, i2, sct, t, f)
+
+        /** Applies effectful transformation of stream elements asynchronously, mapping them in parallel. Preserves chunk boundaries.
+          *
+          * @param parallel
+          *   Maximum number of elements to transform in parallel at a time
+          * @param bufferSize
+          *   Size of buffer used to mediate
+          * @param f
+          *   Asynchronous transformation of stream elements
+          */
+        def mapPar[V2, S2](parallel: Int, bufferSize: Int = defaultAsyncStreamBufferSize)(f: V => V2 < (Abort[E] & Async & S2))(
+            using
+            t1: Tag[Emit[Chunk[V]]],
+            t2: Tag[Emit[Chunk[V2]]],
+            t3: Tag[V2],
+            i1: Isolate.Contextual[S & S2, IO],
+            i2: Isolate.Stateful[S & S2, Abort[E] & Async],
+            ev: SafeClassTag[E | Closed],
+            frame: Frame
+        ): Stream[V2, Abort[E] & Async & S & S2] =
```

Comment:
I still sometimes get impressed to see Kyo's effects all playing well together like here. Cool work!